streamer/src/streamer.rs

```diff
@@ -362,7 +362,7 @@ fn recv_send(
         let data = pkt.data(..)?;
         socket_addr_space.check(&addr).then_some((data, addr))
     });
-    batch_send(sock, &packets.collect::<Vec<_>>())?;
+    batch_send(sock, packets.collect::<Vec<_>>())?;
```

Comment:
why are we still doing collect here if we can pass iterators now?

Reply:
We need an `ExactSizeIterator`. A `filter_map` is not an `ExactSizeIterator` because the number of items filtered out are not known.